justfile

```diff
@@ -44,12 +44,26 @@ test-spec: test-spec-h2
 test-ignored:
 	cargo test --features=cli,telemetry,compression,http-full,proxy-full,tcp,rustls --workspace -- --ignored
 
-qq: lint check clippy doc extra-checks
+qq: ensure-dev-dependencies lint check clippy doc extra-checks
 
 qa: qq test
 
 qa-full: qa hack test-ignored fuzz-60s check-links
 
+ensure-dev-dependencies:
+	@just check-dev-dependencies || just install-dev-dependencies
+
+install-dev-dependencies:
+	cargo install cargo-sort cargo-hack cargo-expand cargo-machete
+
+check-dev-dependencies:
+	#!/usr/bin/env bash
+	installed=$(cargo install --list)
+	echo "$installed" | grep -q "cargo-sort"
+	echo "$installed" | grep -q "cargo-hack"
```

Comment:
I would honestly make these commands either generic or just inline them. These auto-installs I like, but they should be part of the cmd where they are used.

E.g. sort should be part of sort, hack part of hack, etc... Otherwise individual commands will still fail. And similar you now install stuff like hack even though that's only used when doing qa-full, etc.